### ISSUE TITLE
Upgrade d3.js

### ISSUE THREAD

**Issue by EliotRagueneau** [MICommunity/ComplexViewer#184]
I'm currently trying to upgrade our front-end stack on IntAct, and in order to reduce the polyfill.js size, I would like to target es2015 as js level. 
However, by doing so, Complex Portal throws an error because of its d3.js version, which dates from May 2016, and apparently doesn't support es2015.
I hope that if you update the d3.js version to LTS (version 7.1.1), it should be able to support it.
I know there are a load of changes between v. 3.5.17 and v. 7.1.1 so I don't expect it to be upgraded really soon, but it would be nice to be upgraded at some point.

Thanks for your nice support, as always,
Eliot

**Comment by colin-combe** [comment]
yes, it would be good to upgrade the d3 version in my projects but, as you know... "loads of changes"

there maybe aren't that many references to d3 in ComplexViewer - my IDE counted 75 occurrences of "d3." of which 41 were "d3.select". But it's still not going to be upgraded really soon.

I don't know if something like this would work, but maybe you could target es2015 by adding the d3 lib to the project and adding import statements for this local file (i.e. make the d3 lib part of the project code and get babel to deal with it also).

**Comment by EliotRagueneau** [comment]
Thanks a lot for having deployed the version 2.2.0, I just noticed that when downloaded we have the whole development folder and not just the dist forlder, not sure it's normal or if it is my changes have broken

**Comment by colin-combe** [comment]
>  not sure it's normal or if it is my changes have broken

i think its always been like that so not something you've broken

i don't know whats considered best practice / "normal".... i thought it was ok to have whole development folder downloaded but if you think not we can look at changing it

I did notice another bug with complex viewer whilst messing around with it. Not something broken by your changes but something that's been there for ages. I'll make a GH issue...
